src/fm/macros.clj

```diff
@@ -6,15 +6,22 @@
   "The `fm` macro."
   [args-form & body]
   (let [ns-sym (symbol (str *ns* "/" (gensym "fm__")))]
-    (utils/fm-form {:fm/sym       ns-sym
-                    :fm/args-form args-form
-                    :fm/body      body})))
+    (first
+     (utils/fm-form
+      {:fm/sym       ns-sym
+       :fm/args-form args-form
+       :fm/body      body}))))
 
 (defmacro defm
   "The `defm` macro."
   [sym args-form & body]
-  (let [ns-sym (symbol (str *ns* "/" sym))
-        form   (utils/fm-form {:fm/sym       ns-sym
-                               :fm/args-form args-form
-                               :fm/body      body})]
-    `(def ~sym ~form)))
+  (let [ns-sym   (symbol (str *ns* "/" sym))
+        form     (utils/fm-form
+                  {:fm/sym       ns-sym
+                   :fm/args-form args-form
+                   :fm/body      body})
+        metadata (last form)
+        form     (first form)]
+    (if (not (empty? metadata))
+      (list `def (with-meta sym metadata) form)
```

Comment:
This was the only way I could figure out how to apply the extra metadata to the symbol and still get the symbol unquoted. If you look at the definition of `defn` this is how they are doing it as well.

I tried to write a version where I did the `if not empty?` in the let bindings to redefine `sym` above and just use the `(list `def sym form)` but it didn't work and I don't know why. 😄

Reply:
I stole this from you, because I couldn't figure out how to get backtick to properly handle something like `(def ^~(do {:k1 :v1}) ,,,)` no matter what I tried.